pkg/plugins/runtime/k8s/webhooks/resourceadmissionchecker.go

```diff
@@ -78,6 +83,42 @@ func (c *ResourceAdmissionChecker) isPrivilegedUser(allowedUsers []string, userI
 	return slices.Contains(allowedUsers, userInfo.Username)
 }
 
+func (c *ResourceAdmissionChecker) validateLabels(r core_model.Resource) *admission.Response {
+	if c.Mode != core.Global {
+		zoneTag, ok := r.GetMeta().GetLabels()[mesh_proto.ZoneTag]
+		if ok && zoneTag != c.ZoneName {
+			return resourceLabelsNotAllowedResponse(mesh_proto.ZoneTag, c.ZoneName, zoneTag)
+		}
+	}
+
+	return c.validatePolicyRole(r)
+}
+
+func (c *ResourceAdmissionChecker) validatePolicyRole(r core_model.Resource) *admission.Response {
+	ns, ok := r.GetMeta().GetLabels()[mesh_proto.KubeNamespaceTag]
+	if !ok {
+		return nil
+	}
+	policy, ok := r.GetSpec().(core_model.Policy)
+	if !ok {
```

Comment:
we can check r.Descriptor somewhere earlier in the code, to avoid handling cast errors

Reply:
changed